StripePaymentSheet/StripePaymentSheet/Source/Internal/API Bindings/v1-elements-sessions/ElementsCustomer.swift

```diff
@@ -17,27 +17,69 @@ struct ElementsCustomer: Equatable, Hashable {
 
     /// Helper method to decode the `v1/elements/sessions` response's `customer` hash.
     /// - Parameter response: The value of the `customer` key in the `v1/elements/sessions` response.
-    public static func decoded(fromAPIResponse response: [AnyHashable: Any]?) -> ElementsCustomer? {
+    public static func decoded(
+        fromAPIResponse response: [AnyHashable: Any]?,
+        enableLinkInSPM: Bool
+    ) -> ElementsCustomer? {
+        guard let response else {
+            return nil
+        }
+
+        let paymentMethods = Self.parsePaymentMethods(
+            from: response,
+            enableLinkInSPM: enableLinkInSPM && FeatureFlags.linkPMsInSPM.isEnabled
+        )
+
         // Required fields
-        guard let response,
-              let paymentMethodsArray = response["payment_methods"] as? [[AnyHashable: Any]],
+        guard let paymentMethods,
               let customerSessionDict = response["customer_session"] as? [AnyHashable: Any],
               let customerSession = CustomerSession.decoded(fromAPIResponse: customerSessionDict)
         else {
             return nil
         }
 
+        // Optional
+        let defaultPaymentMethod = response["default_payment_method"] as? String
+        return ElementsCustomer(
+            paymentMethods: paymentMethods,
+            defaultPaymentMethod: defaultPaymentMethod,
+            customerSession: customerSession
+        )
+    }
+
+    private static func parsePaymentMethods(from response: [AnyHashable: Any], enableLinkInSPM: Bool) -> [STPPaymentMethod]? {
+        guard let paymentMethodsArray = selectPaymentMethods(from: response, enableLinkInSPM: enableLinkInSPM) else {
+            return nil
+        }
+
         var paymentMethods: [STPPaymentMethod] = []
-        for paymentMethodJSON in paymentMethodsArray {
-            let paymentMethod = STPPaymentMethod.decodedObject(fromAPIResponse: paymentMethodJSON)
-            if let paymentMethod = paymentMethod {
-                paymentMethods.append(paymentMethod)
+        for json in paymentMethodsArray {
+            if enableLinkInSPM {
+                if let paymentMethodWithLinkDetails = PaymentMethodWithLinkDetails.decodedObject(fromAPIResponse: json) {
```

Comment:
Can we fall back to parsing the payment methods if this is missing? We ~know that those will always exist.

Reply:
Oh I see, that's basically what you're doing